tests/integration-tests/utils.py

```diff
@@ -66,6 +66,22 @@ def retrieve_cfn_outputs(stack_name, region):
         raise
 
 
+@retry(wait_exponential_multiplier=500, wait_exponential_max=5000, stop_max_attempt_number=5)
+def retrieve_cfn_resources(stack_name, region):
+    """Retrieve CloudFormation Stack Resources from a given stack."""
+    logging.debug("Retrieving stack resources for stack {}".format(stack_name))
+    try:
+        cfn = boto3.client("cloudformation", region_name=region)
+        stack_resources = cfn.list_stack_resources(StackName=stack_name)["StackResourceSummaries"]
+        resources = {}
+        for resource in stack_resources:
+            resources[resource.get("LogicalResourceId")] = resource.get("PhysicalResourceId")
+        return resources
+    except Exception as e:
+        logging.warning("Failed retrieving stack resources for stack {} with exception: {}".format(stack_name, e))
```

Comment:
Just a note: Python 2.6 and Python 3.0 require that you number the placeholders:
* _Changed in version 2.7: The positional argument specifiers can be omitted for str.format() and unicode.format(), so '{} {}' is equivalent to '{0} {1}', u'{} {}' is equivalent to u'{0} {1}'._
* _Changed in version 3.1: The positional argument specifiers can be omitted for str.format(), so '{} {}'.format(a, b) is equivalent to '{0} {1}'.format(a, b)._

Reply:
Thanks 👍 In any case the integration tests code is not compatible with older python versions for several other features.